python/az/aro/azext_aro/_validators.py

```diff
@@ -64,6 +64,22 @@ def validate_cluster_resource_group(cmd, namespace):
                 namespace.cluster_resource_group)
 
 
+def validate_disk_encryption_set(cmd, namespace):
+    if namespace.disk_encryption_set is not None:
+        if not is_valid_resource_id(namespace.disk_encryption_set):
+            raise InvalidArgumentValueError(
+                "Invalid --disk-encryption-set '%s', has to be a resource ID." %
+                namespace.disk_encryption_set)
+
+        desid = parse_resource_id(namespace.disk_encryption_set)
+        compute_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_COMPUTE)
+        try:
+            compute_client.disk_encryption_sets.get(resource_group_name=desid['resource_group'],
+                                                    disk_encryption_set_name=desid['name'])
+        except CloudError as err:
+            raise InvalidArgumentValueError(err.message) from err
```

Comment:
Should we indicate that this is related to `--disk-encryption-set`? Like above: `Invalid --disk-encryption-set: blah-blah`.

Otherwise (_I think_) it will not be clear what argument is invalid.

Reply:
`validate_subnet` probably has the same issue.

I'm happy for this to be addressed in a separate PR.

Reply:
Added a better message.